Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/PreserveOutputsTests.cs

```diff
@@ -291,6 +291,55 @@ public void PreserveOutputsTestWithWhitelist()
             XAssert.AreEqual(CONTENT, File.ReadAllText(ArtifactToString(outputUnpreserved)));
         }
 
+        [Fact]
+        public void PreserveOutputsTestWithTrustLevel()

```

Comment:
Please add unit tests where you modify `Configuration.Sandbox.UnsafeSandboxConfigurationMutable.PreserveOutputsTrustLevel` from higher to lower and vice versa.